sshdconfig/src/util.rs

```diff
@@ -2,11 +2,41 @@
 // Licensed under the MIT License.
 
 use rust_i18n::t;
+use serde::{Deserialize, Serialize};
+use serde_json::{Map, Value};
 use std::process::Command;
+use tracing::debug;
 use tracing_subscriber::{EnvFilter, filter::LevelFilter, Layer, prelude::__tracing_subscriber_SubscriberExt};
 
 use crate::error::SshdConfigError;
+use crate::parser::parse_text_to_map;
 
+pub struct CommandInfo {
+    /// metadata provided with the command
+    pub metadata: Map<String, Value>,
```

Comment:
Having metadata strongly typed helps it to reject anything it doesn't understand